streams/shared/src/main/scala/zio/stream/ZChannel.scala

```diff
@@ -716,7 +716,64 @@ sealed trait ZChannel[-Env, -InErr, -InElem, -InDone, +OutErr, +OutElem, +OutDon
   final def mapOutZIOParUnordered[Env1 <: Env, OutErr1 >: OutErr, OutElem2](n: Int, bufferSize: Int = 16)(
     f: OutElem => ZIO[Env1, OutErr1, OutElem2]
   )(implicit trace: Trace): ZChannel[Env1, InErr, InElem, InDone, OutErr1, OutElem2, OutDone] =
-    mapOutZIOPar[Env1, OutErr1, OutElem2](n, bufferSize)(f)
+    ZChannel.unwrapScopedWith { scope =>
+      for {
+        input       <- SingleProducerAsyncInput.make[InErr, InElem, InDone]
+        queueReader  = ZChannel.fromInput(input)
+        outgoing    <- Queue.bounded[Exit[Either[Unit, OutDone], OutElem2]](bufferSize)
+        _           <- scope.addFinalizer(outgoing.shutdown)
+        errorSignal <- Promise.make[Unit, Nothing]
+        permits     <- Semaphore.make(n.toLong)
+        failure     <- Ref.make[Cause[OutErr1]](Cause.empty)
+        pull        <- (queueReader >>> self).toPullIn(scope)
+        _ <- pull
+               .foldCauseZIO(
+                 cause =>
+                   failure.update(_ && cause) *>
+                     outgoing.offer(Exit.fail(Left(()))) *>
+                     Exit.failCause(Cause.unit),
+                 {
+                   case Left(outDone) =>
+                     permits.withPermits(n.toLong)(ZIO.unit).interruptible *> outgoing.offer(Exit.fail(Right(outDone)))
+                   case Right(outElem) =>
+                     for {
+                       latch <- Promise.make[Nothing, Unit]
+                       _ <- permits.withPermit {
+                              latch.succeed(()) *>
+                                ZIO.uninterruptibleMask { restore =>
+                                  restore(errorSignal.await).raceFirstAwait(
+                                    restore(f(outElem))
+                                      .catchAllCause(cause => failure.update(_ && cause) *> Exit.failCause(Cause.unit))
```

Comment:
I believe you can add this to the `foldCauseZIO` below

Reply:
Ah yeah my bad, I reviewed this in the web UI and missed that the method was on the RHS effect